packages/ckeditor5-fullscreen/src/handlers/abstracteditorhandler.ts

```diff
@@ -650,4 +665,33 @@ export default class AbstractEditorHandler {
 			);
 		}
 	}
+
+	/**
+	 * Saves the scroll positions of all ancestors of the given element.
+	 */
+	private _saveAncestorsScrollPositions( domElement: HTMLElement ): void {
+		let element = domElement.parentElement;
+
+		if ( !element ) {
+			return;
+		}
+
+		while ( element ) {
+			const overflow = element.style.overflowY || global.window.getComputedStyle( element ).overflowY;
+
+			if ( overflow === 'auto' || overflow === 'scroll' ) {
```

Comment:
Why do we check only for these overflow values? Is it because otherwise element can't be scrolled (and/or the `scrollTop` and `scrollLeft` values are always `0`) or is it because it produced quirks on non-scrollable / overflow:hidden elements if we tried to scroll them?

Although in general I agree with this code, I wonder if we are not accidentally missing something by this kind of "filtering".

BTW, shouldn't we also check `overflowX`? It can be set independently from `overflowY`, so it seems we should check both in case we have applications with horizontal scrolls, and the parent was scrolled horizontally only.

Reply:
I based on https://github.com/ckeditor/ckeditor5/blob/6eca87a05212e01c7067426d41dc21a3a73543af/packages/ckeditor5-utils/src/dom/findclosestscrollableancestor.ts#L18-L39

with an extension that we're also interested in window (<html> element) scroll, where the difference is it can be scrollable even without overflow property set.

Out of 5 possible keyword values: `visible`, `hidden`, `clip`, `scroll` and `auto` - only the last two allow for scrolling.

As for the `overflowX` part, I agree it's possible - I'll enhance the checks.